src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -1565,7 +1567,8 @@ void EmitOr()
                     // character-by-character while respecting the culture.
                     if (!caseInsensitive)
                     {
-                        using (EmitBlock(writer, $"if (!global::System.MemoryExtensions.StartsWith({textSpanLocal}.Slice({textSpanPos}), {Literal(node.Str)}))"))
+                        string sourceSpan = textSpanPos > 0 ? $"{textSpanLocal}.Slice({textSpanPos})" : textSpanLocal;
```

Comment:
Can't we do this as well on the else block bellow when doing caseInsensitive?

Reply:
You mean for the `{textSpanPos} + {i}`?  Yeah, that could be changed to be `{i}` if textSpanPos == 0.  It won't actually matter from a perf perspective as the `0 + i` will be simplified by the C# compiler to just `i`, but from a readability perspective it'd be nice to clean up. I'll do that, thanks.